holoviews/core/util.py

```diff
@@ -1118,18 +1118,20 @@ def max_extents(extents, zrange=False):
         for lidx, uidx in inds:
             lower = [v for v in arr[lidx] if v is not None and not is_nan(v)]
             upper = [v for v in arr[uidx] if v is not None and not is_nan(v)]
-            if lower and isinstance(lower[0], datetime_types):
-                extents[lidx] = np.min(lower)
-            elif any(isinstance(l, str) for l in lower):
-                extents[lidx] = np.sort(lower)[0]
-            elif lower:
-                extents[lidx] = np.nanmin(lower)
-            if upper and isinstance(upper[0], datetime_types):
-                extents[uidx] = np.max(upper)
-            elif any(isinstance(u, str) for u in upper):
-                extents[uidx] = np.sort(upper)[-1]
-            elif upper:
-                extents[uidx] = np.nanmax(upper)
+            if lower:
+                if any(isinstance(l, str) for l in lower):
+                    extents[lidx] = sorted(lower, key=str)[0]
```

Comment:
Somehow, this got lost in the review I made. I had the feeling `max_extents()` was designed to deal with homogeneous types. In this case, when it runs the new test, it sorts a datetime and a string. Is this a valid thing to do?

Reply:
It is very speculative. But if the original code had been something like `if isinstance(lower[0], str)`, like the datetime, I would likely have agreed with you, but the current code looks for any instances of str.